sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -446,9 +446,9 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    * @since 2.1.1
    */
   override def recoverPartitions(tableName: String): Unit = {
```

Comment:
It's probably time to figure out the public Scala API for multi-catalogs: Shall we have one single `CatalogImpl` instance, and all its methods support catalog? Or shall we have one `CatalogImpl` instance per catalog?

The change here makes `recoverPartitions` to support multi-catalogs.

Reply:
Currently, other methods in `CatalogImpl` can work with both v1 and v2 tables already. Look at `cacheTable()`, `isCached()`, `refreshTable()`. I don't see much difference between those methods and `recoverPartitions()`.

And the focus of this PR is to re-use new resolution framework, and to have consistent error message. Not taking into account that recovering partition of v2 tables is not supported at the moment.

In general, `CatalogImpl` has a lot of pretty specific to v1 catalog implementation things.

Reply:
ah since it's already the case, let's move forward then.